src/Common/FileSystem.cpp

```diff
@@ -19,11 +19,43 @@
  */
 
 #include <Common/FileSystem.h>
+
 #include <cstddef>
+#include <algorithm>
+#include <array>
 
 namespace openblack
 {
 
+std::string FileSystem::FixPath(const std::string& path)
+{
+	std::string result = path;
+
+	constexpr std::array<std::string_view, 3> caseFixTable = {
+		"\\Data\\",
+		"\\Landscape\\",
+		"\\Multi_Player\\",
+	};
+	for (auto& pattern : caseFixTable)
+	{
+		auto foundIter = std::search(result.cbegin(), result.cend(), pattern.cbegin(), pattern.cend(),
```

Comment:
The function is doing the inverse operation.
Results needs to not be all upper case.
The paths come in all upper case, such as `.\DATA\LANDSCAPE\MULTI_PLAYER` and the function replaced them with the actual paths which are `.\Data\Landscape\Multi_Player`, then it will fix the slashes depending on the platform.
This allows finding the actual files based on how they're installed on the system (non-upper case).
I should have made that clear in the commit message.

Reply:
Ah ok there is a regex lib in C++. Not used it much

Reply:
If these get somewhat complex I would use the regex lib. For now, this is just replacing strings with same length strings.